### PR TITLE
Fixed notes color from note block being wrong in 1.12.2 -> 1.13

### PR DESCRIPTION
**Pull request by FlorianMichael**
Closes https://github.com/ViaVersion/ViaBackwards/issues/679

### CODE REVIEW

File: common/src/main/java/com/viaversion/viabackwards/protocol/protocol1_12_2to1_13/packets/BlockItemPackets1_13.java [hunk 1]
```diff
@@ -189,6 +203,13 @@ public void register() {
                     int blockState = wrapper.read(Type.VAR_INT);
                     Position position = wrapper.get(Type.POSITION1_8, 0);
 
+                    // Note block special treatment
+                    if (blockState >= 249 && blockState <= 748) { // Note block states id range
+                        wrapper.user().get(NoteBlockStorage.class).storeNoteBlockUpdate(position, blockState);
+                        wrapper.cancel();
+                        return;
```
Comment (kennytv) [inline common/src/main/java/com/viaversion/viabackwards/protocol/protocol1_12_2to1_13/packets/BlockItemPackets1_13.java:210]: The storage checkAndStore still needs to be called and you can't just cancel the block change from air

**Review by kennytv — CHANGES_REQUESTED** [review]

File: common/src/main/java/com/viaversion/viabackwards/protocol/protocol1_12_2to1_13/storage/NoteBlockStorage.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import com.viaversion.viaversion.util.Pair;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class NoteBlockStorage implements StorableObject {
+
+    private static final int MAX_NOTE_ID = 24;
+
+    private final Map<Position, Integer> noteBlockUpdates = new ConcurrentHashMap<>();
```
Comment (kennytv) [inline common/src/main/java/com/viaversion/viabackwards/protocol/protocol1_12_2to1_13/storage/NoteBlockStorage.java:31]: This doesnt need to be concurrent, use Object2IntMap instead

File: common/src/main/java/com/viaversion/viabackwards/protocol/protocol1_12_2to1_13/storage/NoteBlockStorage.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+package com.viaversion.viabackwards.protocol.protocol1_12_2to1_13.storage;
+
+import com.viaversion.viaversion.api.connection.StorableObject;
+import com.viaversion.viaversion.api.minecraft.Position;
+import com.viaversion.viaversion.util.Pair;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class NoteBlockStorage implements StorableObject {
```
Comment (kennytv) [inline common/src/main/java/com/viaversion/viabackwards/protocol/protocol1_12_2to1_13/storage/NoteBlockStorage.java:27]: Needs to be cleared on respawn

**Review by kennytv — CHANGES_REQUESTED** [review]

File: common/src/main/java/com/viaversion/viabackwards/protocol/protocol1_12_2to1_13/packets/BlockItemPackets1_13.java (hunk 1 above)
Comment (FlorianMichael) [reply common/src/main/java/com/viaversion/viabackwards/protocol/protocol1_12_2to1_13/packets/BlockItemPackets1_13.java]: Fixed

**Review by kennytv — APPROVED** [review]

**Pull request merged**